lib/tasks_private/openstack.rake

```diff
@@ -15,14 +15,10 @@ namespace :vcr do
     end
   end
 
-  desc 'Rerecord all of VCR cassettes'
-  task :rerecord => :environment do
+  namespace :environment do
     require 'fog/openstack'
 
     host, port, username, password = YAML.load_file("config/secrets.yml").dig("test", "openstack").values_at("hostname", "port", "userid", "password")
```

Comment:
The reason the rake tasks are failing is because this code is right in the namespace so it is being evaluated when the rake file is loaded rather than just when it is being called.

You can create methods to accomplish the same thing like:
```ruby
namespace :environment do
    def secrets_yml
      @secrets_yml ||= YAML.load_file("config/secrets.yml").dig("test", "openstack")
    end

    def fog_connect_opts
      host, port, username, password = secrets_yml.values_at("hostname", "port", "userid", "password")

      {
        :provider               => "OpenStack",
        :openstack_auth_url     => "https://#{host}:#{port}/v3/",
        etc...
      }
    end

    def compute_client
      require 'fog/openstack'
      Fog::Compute.new(fog_connect_opts)
    end

    def network_client
      require 'fog/openstack'
      Fog::Network.new(fog_connect_opts)
    end
```
```